pkg/webhook/convert.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// volumeSnapshotInfoAnnotationName is the name of the annotation
+	// that is used when converting data from the v1beta2 to the v1beta1
+	// API to make the conversion reversible.
+	volumeSnapshotInfoAnnotationName = "groupsnapshot.storage.kubernetes.io/volume-snapshot-info-list"
+)
+
+func convertGroupSnapshotCRD(obj *unstructured.Unstructured, toVersion string) (*unstructured.Unstructured, metav1.Status) {
+	klog.V(2).Info("converting crd")
+
+	convertedObject := obj.DeepCopy()
+	fromVersion := obj.GetAPIVersion()
+	kind := obj.GetKind()
+
+	if toVersion == fromVersion {
+		return nil, statusErrorWithMessage("conversion from a version to itself should not call the webhook: %s", toVersion)
+	}
+
+	switch obj.GetAPIVersion() {
+	case "groupsnapshot.storage.k8s.io/v1beta1":
+		switch toVersion {
+		case "groupsnapshot.storage.k8s.io/v1beta2":
+			switch kind {
+			case "VolumeGroupSnapshot":
+			case "VolumeGroupSnapshotClass":
```

Comment:
Remove these cases, the webhook should not convert VolumeGroupSnapshots and VolumeGroupSnapshotClasses. Throw `unexpected conversion kind %q` below.

Reply:
Removed.